src/Development/IDE/Core/Shake.hs

```diff
@@ -346,6 +348,11 @@ shakeOpen getLspId eventer withProgress withIndefiniteProgress logger debouncer
         publishedDiagnostics <- newVar mempty
         positionMapping <- newVar HMap.empty
         let restartShakeSession = shakeRestart ideState
+        progressQueue <- newTVarIO KickCompleted
+        let progressUpdate = atomically . writeTVar progressQueue
+        when reportProgress $
+            void $ async $ progressThread progressQueue inProgress
```

Comment:
Afaict this thread will keep running even if you call `shakeShut`. While I think this probably doesn’t matter much for ghcide since we usually only call this much it matters when you use ghcide as a library and might run multiple sessions within a single process.

Reply:
good point, will fix